docs/blueprints/EP035.rst

```diff
@@ -0,0 +1,231 @@
+The kytos event bus provides only one way communication,
+making it unsuitable for this task.
+
+#############
+Specification
+#############
+
+The following is a set of proposed changes for
+how to handle interruptions in the future,
+in order to have visibility on there causes and effects.
+
+Kytos Core Changes
+##################
+
+In order for services to identify which interruptions are affecting the devices
+they use, a mechanism needs to be added to the device representation for
+identify interruption causes.
+
+
+Extend ``status_funcs`` to produce a ``status_reason``
+------------------------------------------------------
+
+Currently, when an object is queried about its ``status``,
+the reason for that status is unknown.
+What we propose here is adding a new property to
+``GenericEntity`` called ``status_reason``,
+and adding in ``status_reason_func``
+to produce values for ``status_reason``.
+The produced ``status_reason`` will be a set of strings,
+so all possible reasons a device might be disable
+can be easily represented.
+
+Example usage:
+
+.. code-block:: python3
+
+    from kytos.core.switch import Switch
+    maintenance_switches = set()
+
+    def switch_maintenance_status(switch):
+        if switch.id in maintenance_switches:
+            return EntityStatus.DOWN
+        return EntityStatus.UP
+
+    Switch.register_status_func('maintenance', switch_maintenance_status)
+
+    def switch_maintenance_reason(switch):
+        if switch.id in maintenance_switches:
+            return {'maintenance'}
+        return set()
+
+    Switch.register_status_reason_func('maintenance', switch_maintenance_reason)
+
+
+New Topology REST Endpoints
+###########################
+
+The following set of new endpoints are intended to allow
+for previewing the affects of adding in interruptions.
+
+
+REST Endpoint ``GET /topology/v3/preview_interruption``
+-------------------------------------------------------
+
+For all known service providers,
+previews affected if the specified interruption occurs now.
+
+Expected Input:
+
+.. code-block:: json
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+
+Expected Response:
+
+.. code-block:: json
+
+    {
+        <provider name str>: [<service name str>],
+    }
+
+Side Effects:
+
+- None
+
+Notes:
+
+- Should support multiple key-value pairs with the same key in the query parameters.
+
+
+Endpoint: ``GET <url for previewing>``
+--------------------------------------
+
+For the given service provider,
+previews services affected if the specified interruption occurs now.
+
+Expected Input:
+
+.. code-block:: json
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Expected Output:
+
+.. code-block:: json
+
+    [
+        {
+            'id': <service name str>,
```

Comment:
In the original mw blueprint, it was also mentioned a desire to also have information about which user is associated to that service. Do we need to provide a richer reponse model here?

`{"id": <id>, "name": <service_name>, "owner": <str|null>}`, and then in the future potentially "username" and "email"?

Reply:
From what I understand it would be nice, but not required. I've made including such information optional for now, just requiring at least an id.

Reply:
Right, this part on the blueprint we're good since at both id and owner got covered, so we wouldn't need more effort on `maintenance`. Did you also confirm this with @italovalcy? If that was the understanding we can close this thread, otherwise we might need to create a new issue on mef_eline to support `owner: str on POST /v2/evc` cc'ing @Ktmi

Reply:
Got confirmation from @italovalcy. Have it remain optional for now, as its unlikely it will be used in the short term.